experimental/iterators/README.md

```diff
@@ -273,3 +273,55 @@ the sequence-based relational model), neither the order of the rows nor that of
 the columns matter, (3) the elements of relations may be strings or similar
 variable-length data types, and (4) nested relational algebra also allows for
 structured element types (which may have variable length).
+
+## Basic Concepts
+
+This section defines the central concepts of this dialect and relates them to
+those introduced in the [Background](#background) section.
+
+* **Stream**: A collection of elements of a particular type that (1) is ordered
+  and (2) can only be iterated over in that order one element at the time.
+
+  Stream is the main data type that iterators consume and produce.
+
+  Ideally, there should be no restriction on the type of the elements and it
+  seems like it is possible to achieve that ideal (since Python achieves it).
```

Comment:
I am not sure I would claim that Python achieves this really. AFAIK, Python just tries to do it and if it works cool, otherwise fail. A aot compiled project like this might need stronger assumptions/regulations than that.